### ISSUE TITLE
actions._close(bufnr, true) exits insert mode

### ISSUE THREAD

**Issue by ikerurda** [nvim-telescope/telescope.nvim#1599]
### Description

I have noticed that plugins that change from one picker to another or even builtin pickers (`:Telescope builtin`) quit insert mode when not intended.

### Neovim version

```markdown
NVIM v0.7.0-dev+22d7dd2ae
Build type: Release
```

### Operating system and version

macOs 12.1

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - WARNING: nvim-treesitter not found.

## Checking external dependencies
  - OK: rg: found ripgrep 12.1.1
  - OK: fd: found fd 8.3.0

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided
```

### Steps to reproduce

1. nvim -nu minimal.lua
2. :Telescope builtin
3. Select a builtin

### Expected behavior

I was expecting to stay in insert mode

### Actual behavior

Vim mode is changed to Normal

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by LinArcX** [comment]
I have the same issue when i was writing my plugin:
https://gist.github.com/LinArcX/1d46cf87a0d55f5313060d4581cebb17

**Comment by Conni2461** [comment]
This is happening due to change in neovim upstream that affected the prompt buffer. It still works in 0.6.0 and we are working on a fix in #1600
